src/kagglehub/clients.py

```diff
@@ -117,6 +117,7 @@ def post(self, path: str, data: dict) -> dict:
             auth=self._get_http_basic_auth(),
             timeout=(DEFAULT_CONNECT_TIMEOUT, DEFAULT_READ_TIMEOUT),
         ) as response:
+            response.raise_for_status()
```

Comment:
the `process_post_response` already checks for errors. why do you think we should add the raise_for_status ?

Reply:
The next line `response_dict = response.json()` throws if there is no response